manifests/interface.pp

```diff
@@ -243,4 +245,22 @@
       fail("provider ${provider} not supported")
     }
   }
+  if 'wireguard_pubkeys' in $facts {
+    if $interface in $facts['wireguard_pubkeys'] {
```

Comment:
Is there a benefit for checking if `wireguard_pubkeys` is in `$facts` and not directly using `if $interface in $facts['wireguard_pubkeys']`?

Reply:
`wireguard_pubkeys` fact is not present in facts on first run , so we cannot check is `$interface` is present in it